gnocchi/rest/api.py

```diff
@@ -1692,19 +1693,15 @@ def validate_qs(start, stop, granularity, needed_overlap, fill):
                         "reason": six.text_type(e)})
 
     if fill is not None:
-        if granularity is None:
-            abort(400, {"cause": "Argument value error",
-                        "detail": "granularity",
-                        "reason": "Unable to fill without a granularity"})
-        if fill != "null":
+        if fill not in ["null", "filtered-null"]:
             try:
                 fill = float(fill)
             except ValueError:
                     abort(400,
                           {"cause": "Argument value error",
                            "detail": "fill",
-                           "reason": "Must be a float or \'null\', got '%s'" %
-                           fill})
+                           "reason": "Must be a float, \'filtered-null\' "
+                           "or \'null\', got '%s'" % fill})
```

Comment:
maybe we could use a voluptuous schema at this point.

Reply:
done